setup.cfg

```
@@ -18,5 +18,46 @@ formats = zip
 universal = 1
 
 [metadata]
-license_file = LICENSE
+name = setuptools
 version = 41.0.1
+description = Easily download, build, install, upgrade, and uninstall Python packages
+author = Python Packaging Authority
+author_email = distutils-sig@python.org
+long_description = file: README.rst
+long_description_content_type = text/x-rst; charset=UTF-8
+license_file = LICENSE
+keywords = CPAN PyPI distutils eggs package management
```

@pganssle I'm pretty sure this is parsed as a string now while it should be a list
```suggestion
keywords =
    CPAN
    PyPI
    distutils
    eggs
    package management
```

---

@benoit-pierre ^

---

See discussion above: https://github.com/pypa/setuptools/pull/1761#issuecomment-492663319  and https://github.com/pypa/setuptools/pull/1761#issuecomment-492666140.

---

Oh... I'm relying on the list interface in my other projects tho

---

If you're putting them on multiple lines, you're okay then. But `keyword1, keyword2` won't generate the right metadata.

---

Ah... I multiline them, yes.

---

Wait, no, if you're putting the keyword on multiple lines, it should not work (generating a comma separated list in the final PKG-INFO). Does PyPI handle your projects metadata correctly?

---

Yes, it does. Which probably means that the de-facto implementations may differ from the standard...

---

Yeah, it's not correct, I just tested with https://test.pypi.org/project/plover-plugins-manager/:
```shell
$ wget --quiet -O - https://test.pypi.org/pypi/plover-plugins-manager/json | json_pp | grep keyword
      "keywords" : "plover,plover_plugin",
```

---

Look:
https://github.com/sanitizers/octomachinery/blob/f056cda/setup.cfg#L65-L73
https://pypi.org/project/octomachinery/

---

```console
➜ wget --quiet -O - https://test.pypi.org/pypi/octomachinery/json | json_pp | grep keyword
      "keywords" : "Bot,Framework,Framework for writing GitHub Apps,GitHub,GitHub Actions,GitHub API,GitHub Apps,GitHub Checks API",
```

---

So? How do you know the keywords are correctly handled? In both case the display on the project page will be the same.

---

Well, I mean that on the web page it's a list as expected.

---

```python
from distlib.index import PackageIndex
pki = PackageIndex()
pki.search({'keyword': 'Framework for writing GitHub Apps'})
pki.search({'keyword': 'Framework for writing GitHub Apps'}).keys()
sorted([m['name'] for m in  pki.search({'keyword': 'Framework for writing GitHub Apps'})])
'octomachinery' in {m['name'] for m in  pki.search({'keyword': 'Framework for writing GitHub Apps'})}
=> False
```

---

Update code above (was using `keyword` instead of `keywords`. Anyway, the JSON metadata is different, and the PKG-INFO is still not PEP 314 compliant, but looks like PyPI work around it... *sigh*

---

*sigh* :)

---

According to the [spec](https://setuptools.readthedocs.io/en/latest/setuptools.html#metadata), `keywords` should be a comma separated list in `setup.cfg`. My original PR was formatted like this but i removed the commas by request of the team.

How should we move on with this issue now?

---

I just checked the code. Setuptools actually [does support](https://github.com/pypa/setuptools/blob/master/setuptools/tests/test_config.py#L85) the correct `keywords` as comma separated list and dangling list. It is just that it uses it wrong it's own `setup.cfg`.

---

Yes, it says `list-comma` which is explained above that table:
```
Type names used below:
...
* list-comma - dangling list or string of comma-separated values
```

---

Yet, for readability, I'd prefer a multiline list instead of an inline comma-separated one.

---

Either way this is a complete side-issue. @benoit-pierre Already figured out that it's just a bug in the code *and* how to fix it, so we should fix that bug and then remove the commas.